src/python/gem5/components/boards/abstract_board.py

```diff
@@ -101,20 +112,22 @@ def __init__(
         # Set the processor, memory, and cache hierarchy.
         self.processor = processor
         self.memory = memory
-        self._cache_hierarchy = cache_hierarchy
+        self._cache_hierarchy: Optional[
+            "AbstractCacheHierarchy"
+        ] = cache_hierarchy
         if cache_hierarchy is not None:
-            self.cache_hierarchy = cache_hierarchy
+            self.cache_hierarchy: "AbstractCacheHierarchy" = cache_hierarchy
```

Comment:
This code seems correct, but it's really ugly. I am going to push back a bit on making changes that make the code more difficult to read.

In general, the type checking in python should improve the code's readability, not make it worse.

Reply:
I'll see if there's a more readable way to write this block

Reply:
yeah turns out the annotations aren't necessary in this case, will remove

Reply:
fixed by [the most recent commit](https://github.com/gem5/gem5/pull/1409/commits/b73fe85e54ab679124848013b91444a7ca3aaf37)